### ISSUE TITLE
Recurring events show up multiple times

### ISSUE THREAD

**Issue by qbixx** [transformstudios/statamic-events#54]
We have setup an event that is a weekly events. This event is showing multiple times in the overview. We are using the query tag like below:

`{{ events:upcoming collection="acties" limit="24" collapse_multi_days="true" }}`

Is there a way to only show these repeating events only once? (the most recent one)?

**Comment by edalzell** [comment]
It's a recurring event?

**Comment by qbixx** [comment]
Yes, it's an recurring event (once a week)

**Comment by edalzell** [comment]
What's the use case? Can you show me what ideal output would look for you? (sample screenshot)

**Comment by qbixx** [comment]
We are using the module to show a combination of weekly recurring events and one time events. With the current setup the overview contains mainly the weekly events. They are all duplicates of the original event (see screenshot below):
![Screenshot 2023-02-19 at 13 56 47](https://user-images.githubusercontent.com/1932778/219949532-038ddecf-be32-4fbb-b9ac-88c1297a6676.png)


It would be really nice if there would be an option to remove the duplicate events so that only the most recent recurring event is showing in the overview. When the most recent date for this event has passed it will show still only once for the next date (see screenshot below):
![Screenshot 2023-02-19 at 13 59 10](https://user-images.githubusercontent.com/1932778/219949549-805afddd-5298-4367-a60f-c996b4c7048f.png)

**Comment by edalzell** [comment]
So you want NO duplicate events shown? Essentially a listing of Events (and not occurrences)?

I don't really understand why you'd want that though, how does it help the viewer?

**Comment by qbixx** [comment]
Yes, ideally we don't want to show the duplicate events. This will help the user because otherwise there is bulk of the same event showing in the overview (e.g. every day) while the individual events are pushed backwards.

On our website we have 5 locations that can have their events listed. All of them have events that take place every day. Next to that the locations have about once a week an individual event. In the current setup the daily events will fill up 35 events (5 locations * 7 days) and the individual event will be almost not visible for user that come by for that specific individual event.

**Comment by edalzell** [comment]
So you would want something like `{{ events:next }}` or something like that goes through the events and outputs only events that have a next event and only shows that one occurrence?

**Comment by qbixx** [comment]
Ideally we would like to show the upcoming events, but for events that have recurring dates, only show the most recent event so that there aren't any duplicate events showing.

In the screenshot above, `Test Event 1` will show only once. Even if it will take place every day.

On Feb. 19, it will show: Test Event 1, Test Event 2, Test Event 3 and Test Event 4
On Feb. 20, it will show: Test Event 1, Test Event 2, Test Event 3 and Test Event 4
On Feb. 21, it will show: Test Event 1, Test Event 3 and Test Event 4
On Feb. 22, it will show: Test Event 1 and Test Event 4
On Feb. 23, it will show: Test Event 1

**Comment by edalzell** [comment]
OK, so if I add an `event` param to the `upcoming` tag, that would allow you to do that, I believe, because you can then use a normal `collection` tag to loop through events, calling `upcoming` on each event.

Sound about right @qbixx?

**Comment by qbixx** [comment]
That sounds like a really nice option

**Comment by qbixx** [comment]
@edalzell Do you perhaps have an update on this feature?

**Comment by edalzell** [comment]
> @edalzell Do you perhaps have an update on this feature?

I'll take a look this week to see if I can fit this in.

**Comment by qbixx** [comment]
@edalzell Did you had time to take a look at this feature?

**Comment by edalzell** [comment]
> @edalzell Did you had time to take a look at this feature?

Shoot got busy. I'll put it on my schedule for next week.

**Comment by edalzell** [comment]
@qbixx check out the latest release, you can now do `{{ events:upcoming :event="id" limit="1" }}` to get the next occurrence of a specific event.

**Comment by qbixx** [comment]
@edalzell Thanks! I just updated the module and made the required changes. The events are now showing up correctly and only show once for events that have multiple days. I've added everything in a regular `{{ collection }}` tag.

One thing I can seem to get working correctly with the collection tag is to show the events in the right order. I'm using the code below:

```
{{ collection:acties sort="start_date" }}
    {{ events:upcoming :event="id" limit="1" }}
         {{ partial:components/action_item }}
    {{ /events:upcoming}}
{{ /collection:acties }}
```

This sorts the events by the default `start_date` field. However, a recurring event that is e.g. on Friday would show up before an event with a single date that is on Monday because the start_date field is before the date of the event on Monday.

Is there a way to make sure the events are showing up in the correct order?
